# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libvpl/hotfix-other-archs.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

libvpl-2.14.0/api/vpl/mfxdefs.h:75:6: error: #error Unknown packing
   75 |     #error Unknown packing
      |      ^~~~~

--- libvpl-2.14.0/api/vpl/mfxdefs.h.vanilla	2025-03-20 10:11:37.243289214 +0100
+++ libvpl-2.14.0/api/vpl/mfxdefs.h	2025-03-20 10:13:53.670319933 +0100
@@ -60,7 +60,7 @@
 #define MFX_PACK_BEGIN_USUAL_STRUCT()        MFX_PACK_BEGIN_X(4)
 
 /* 64-bit LP64 data model */
-#if defined(_WIN64) || defined(__LP64__)
+#if defined(_WIN64) || defined(__LP64__) || __SIZEOF_LONG__ == 8
     #define MFX_PACK_BEGIN_STRUCT_W_PTR()    MFX_PACK_BEGIN_X(8)
     #define MFX_PACK_BEGIN_STRUCT_W_L_TYPE() MFX_PACK_BEGIN_X(8)
 /* 32-bit ILP32 data model Windows* (Intel(r) architecture) */
@@ -68,7 +68,7 @@
     #define MFX_PACK_BEGIN_STRUCT_W_PTR()    MFX_PACK_BEGIN_X(4)
     #define MFX_PACK_BEGIN_STRUCT_W_L_TYPE() MFX_PACK_BEGIN_X(8)
 /* 32-bit ILP32 data model Linux* */
-#elif defined(__ILP32__) || defined(__arm__)
+#elif defined(__ILP32__) || defined(__arm__) || __SIZEOF_LONG__ == 4
     #define MFX_PACK_BEGIN_STRUCT_W_PTR()    MFX_PACK_BEGIN_X(4)
     #define MFX_PACK_BEGIN_STRUCT_W_L_TYPE() MFX_PACK_BEGIN_X(4)
 #else
